app/services/katello/registration_manager.rb

```diff
@@ -151,14 +151,16 @@ def lookup_content_view_environment(activation_keys)
 
       def candlepin_consumer_destroy(host_uuid)
         ::Katello::Resources::Candlepin::Consumer.destroy(host_uuid)
+      rescue RestClient::ResourceNotFound
+        Rails.logger.warn(_("Attempted to destroy consumer %s from candlepin, but consumer does not exist in candlepin") % host_uuid)
       rescue RestClient::Gone
-        Rails.logger.error(_("Consumer %s has already been removed") % host_uuid)
+        Rails.logger.warn(_("Consumer %s has already been removed") % host_uuid)
```

Comment:
tiny change since you are already in the area, can this be `"Candlepin consumer %s.."`? just to make extra clear what is happening.